lib/auth/auth_with_roles.go

```diff
@@ -1041,41 +1034,38 @@ func (a *AuthWithRoles) GenerateUserCerts(ctx context.Context, req proto.UserCer
 	}, nil
 }
 
-func (a *AuthWithRoles) CreateSignupToken(user services.UserV1, ttl time.Duration) (token string, e error) {
-	if err := a.action(defaults.Namespace, services.KindUser, services.VerbCreate); err != nil {
-		return "", trace.Wrap(err)
-	}
-	return a.authServer.CreateSignupToken(user, ttl)
-}
-
-func (a *AuthWithRoles) GetSignupTokenData(token string) (user string, otpQRCode []byte, err error) {
+func (a *AuthWithRoles) GetSignupU2FRegisterRequest(token string) (u2fRegisterRequest *u2f.RegisterRequest, e error) {
 	// signup token are their own authz resource
-	return a.authServer.GetSignupTokenData(token)
+	return a.authServer.CreateSignupU2FRegisterRequest(token)
 }
 
-func (a *AuthWithRoles) GetSignupToken(token string) (*services.SignupToken, error) {
-	// signup token are their own authz resource
-	return a.authServer.GetSignupToken(token)
-}
+func (a *AuthWithRoles) CreateResetPasswordToken(ctx context.Context, req CreateResetPasswordTokenRequest) (services.ResetPasswordToken, error) {
```

Comment:
@alex-kovoy This should also be limited so a user can only reset their own password or `tctl` can reset any password.

Reply:
Right now only admin (who has `services.KindUser, services.VerbUpdate` permissions) can do that. Users without this permission are not allowed to reset passwords even their own. 

There is a separate API which allows a user to change own password (Account screen) but this is not the same as reset. 

Given that, what should be done differently in this regard?

Reply:
+ there is a new UI coming which allows admins to reset user passwords from WEB UI.

Reply:
Okay, makes sense.